recipes/gc/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class GcConan(ConanFile):
+    name = "gc"
+    homepage = "https://www.hboehm.info/gc/"
+    description = "The Boehm-Demers-Weiser conservative C/C++ Garbage Collector (libgc, bdwgc, boehm-gc)"
+    topics = ("conan", "gc", "garbage", "collector")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+
+    _autotools_options_defaults = (
+        ("cplusplus",                   False,),
+        ("throw_bad_alloc_library",     True,),
+        ("cord",                        True,),
+        ("threads",                     True,),
+        ("parallel_mark",               True,),
+        ("handle_fork",                 True,),
+        ("thread_local_alloc",          True,),
+        ("threads_discovery",           True,),
+        ("parallel_mark",               True,),
+        ("gcj_support",                 True,),
+        ("java_finalization",           True,),
+        ("sigrt_signals",               False,),
+        ("atomic_uncollectable",        True,),
+        ("gc_debug",                    False,),
+        ("redirect_malloc",             False,),
+        ("disclaim",                    True,),
+        ("large_config",                True,),
+        ("gc_assertions",               False,),
+        ("mmap",                        False,),
+        ("munmap",                      True,),
+        ("dynamic_loading",             True,),
+        ("register_main_static_data",   True,),
+        ("gc_assertions",               False,),
+        ("checksums",                   False,),
+        ("single_obj_compilation",      False,),
+    )
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    for option, default in _autotools_options_defaults:
+        options[option] = [True, False]
+        default_options[option] = default
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if tools.Version(self.version) <= "8.0.4":
+            del self.options.throw_bad_alloc_library
```

Comment:
```suggestion
            del self.options.throw_bad_alloc_library
        if not self.options.cplusplus:
            del self.settings.compiler.libcxx
            del self.settings.compiler.cppstd
```

Both libgc and libcord don't require C++ lib